### PR TITLE
adds a nightly build action and build nightly rc from main

### PR DESCRIPTION
**Pull request by ripienaar**

### CODE REVIEW

**Comment by ripienaar** [comment]
This creates a reuable action for building images, then use that in a new `rc` image, leaving the `nightly` one as is so it wont fail.

If this new build work I will upate the `nightly` tomorrow to use the same reusable action.

**Review by wallyqs — APPROVED** [review]
LGTM

**Comment by bruth** [comment]
Given your last suggestion on Slack, I think `nightly-main` is more intuitive than `rc`.

File: .github/workflows/rc_nightly.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+jobs:
+  rc_release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: ./.github/actions/nightly-release
+        with:
+          go: "1.19"
+          branch: main
+          label: nightly-main
```
Comment (ripienaar) [inline .github/workflows/rc_nightly.yaml:17]: updated as discussed, will merge on green.

**Pull request merged**